### PR TITLE
Horizontal resizing

### PR DESCRIPTION
**Pull request by uninhm**

### CODE REVIEW

**Comment by uninhm** [comment]
#143

File: src/nimdowpkg/windowmanager.nim [hunk 1]
```diff
@@ -542,6 +542,18 @@ proc jumpToUrgentWindow(this: WindowManager) =
   urgentMonitor.setSelectedTags(tagID)
   this.display.warpTo(urgentClient)
 
+proc incWidthDiff(this: WindowManager) =
+  var layout = this.selectedMonitor.taggedClients.findFirstSelectedTag.layout
+  if layout of MasterStackLayout:
+    cast[MasterStackLayout](layout).widthDiff += 10
```
Comment (avahe-kellenberger) [inline src/nimdowpkg/windowmanager.nim:548]: Could you make the `widthDiff` value (10 here) adjustable via the config file?

If it's too much of a hassle, I could get to this in a few days

**Comment by avahe-kellenberger** [comment]
Just as a note, I tested it and it seems to be working very well.

The only edge case I see that need covering, is resizing too far. If you hold `super l` or `super h` too long you'll see Nimdow will crash

File: src/nimdowpkg/windowmanager.nim (hunk 1 above)
Comment (uninhm) [reply src/nimdowpkg/windowmanager.nim]: How do I name that configuration?

Comment (uninhm) [reply src/nimdowpkg/windowmanager.nim]: Maybe `resizeStep`?

**Comment by uninhm** [comment]
Sorry for the double commit, I thought I hadn't committed yet

File: src/nimdowpkg/windowmanager.nim [hunk 2]
```diff
@@ -542,6 +542,23 @@ proc jumpToUrgentWindow(this: WindowManager) =
...
   this.display.warpTo(urgentClient)
 
+template modWidthDiff(this: WindowManager, diff: int) =
+  var layout = this.selectedMonitor.taggedClients.findFirstSelectedTag.layout
+  if layout of MasterStackLayout:
+    var masterStackLayout = cast[MasterStackLayout](layout)
+    let screenWidth = masterStackLayout.calcScreenWidth(this.selectedMonitor.layoutOffset)
+    if (diff > 0 and masterStackLayout.widthDiff < 0) or
+      (diff < 0 and masterStackLayout.widthDiff > 0) or
+      masterStackLayout.calcClientWidth(screenWidth).int - abs(masterStackLayout.widthDiff).int - 10 > 0:
```
Comment (avahe-kellenberger) [inline src/nimdowpkg/windowmanager.nim:552]: I believe the 10 here is still hard-coded but shouldn't be?

File: src/nimdowpkg/wmcommands.nim [hunk 3]
```diff
@@ -27,6 +27,8 @@ type
     wmcDestroySelectedWindow = "destroyselectedwindow",
     wmcToggleFloating = "togglefloating",
     wmcJumpToUrgentWindow = "jumptourgentwindow"
+    wmcIncWidthDiff = "incwidthdiff"
+    wmcDecWidthDiff = "decwidthdiff"
```
Comment (avahe-kellenberger) [inline src/nimdowpkg/wmcommands.nim:31]: I think we should come up with a name that better indicates what the config property is, but I haven't thought of one yet

File: src/nimdowpkg/windowmanager.nim (hunk 2 above)
Comment (uninhm) [reply src/nimdowpkg/windowmanager.nim:552]: You're right

**Comment by dakyskye** [comment]
Works very well :+1: great job! However, I wish there was a way to reset that resize. Probably a keybinding for it, or automatically reset it when all slave windows get closed :man_shrugging:

**Comment by avahe-kellenberger** [comment]
Let's make an issue for it, and implement it in another PR. I want to do a few renames and merge this in today.

**Comment by uninhm** [comment]
Is there something I can do?

**Comment by avahe-kellenberger** [comment]
@uninhm See the comment(s) I left, small changes then I can merge

**Comment by uninhm** [comment]
@avahe-kellenberger done

File: config.default.toml [hunk 4]
```diff
@@ -143,6 +145,14 @@ exec = [
   keys = [ "1", "2", "3", "4", "5", "6", "7", "8", "9" ]
   modifiers = [ "super", "shift", "control" ]
 
+	[controls.incWidthDiff]
+	keys = [ "l" ]
+	modifiers = [ "super" ]
+
+	[controls.decWidthDiff]
+	keys = [ "h" ]
+	modifiers = [ "super" ]
```
Comment (avahe-kellenberger) [inline config.default.toml:154]: Need to changes these as well and we should be good.

Comment (uninhm) [reply config.default.toml:154]: Ok

**Review by avahe-kellenberger — APPROVED** [review]

**Pull request merged**